### PR TITLE
Multi-sub peering initial implementation

### PR DESCRIPTION
**Pull request by deepend-dev**
resolves below -
#206
#259

### CODE REVIEW

File: Cloudmarque.Azure/Public/Common/Get-CmAzResourceName.ps1 [hunk 1]
```diff
@@ -91,6 +91,8 @@
 
 			$regionConvention = $regions.regions | Where-Object { $_.name -Eq $Region }
 
+			$regionConvention ??= $regions.regions | Where-Object { $_.alias -contains $Region }
```
Comment (tm-luke) [inline Cloudmarque.Azure/Public/Common/Get-CmAzResourceName.ps1:94]: Is this needed, with the line above?

File: Cloudmarque.Azure/Public/Deploy/IaaS/networking/vnetPeerings/New-CmAzIaasVnetPeerings.json [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+                "parameters": {
+                    "VnetPeeringsObject": {
+                        "value": "[parameters('VnetPeerings')[copyIndex('VnetPeeringsCopy')]]"
+                    }
+                },
+                "template": {
+                    "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
+                    "contentVersion": "1.0.0.0",
+                    "parameters": {
+                        "VnetPeeringsObject": {
```
Comment (tm-luke) [inline Cloudmarque.Azure/Public/Deploy/IaaS/networking/vnetPeerings/New-CmAzIaasVnetPeerings.json:33]: Remove object from name

File: Cloudmarque.Azure/Public/Deploy/IaaS/networking/vnetPeerings/New-CmAzIaasVnetPeerings.ps1 [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+	$ErrorActionPreference = "Stop"
+
+	try {
+
+		Get-InvocationInfo -CommandName $MyInvocation.MyCommand.Name
+
+        $SettingsObject = Get-Settings -SettingsFile $SettingsFile -SettingsObject $SettingsObject -CmdletName (Get-CurrentCmdletName -ScriptRoot $PSCommandPath)
+
+        $azContext = Get-AzContext
```
Comment (tm-luke) [inline Cloudmarque.Azure/Public/Deploy/IaaS/networking/vnetPeerings/New-CmAzIaasVnetPeerings.ps1:44]: Move below what-if check

File: Cloudmarque.Azure/Public/Deploy/IaaS/networking/vnetPeerings/New-CmAzIaasVnetPeerings.ps1 [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+
+		if ($PSCmdlet.ShouldProcess((Get-CmAzSubscriptionName), "Create Vnet Peerings")) {
```
Comment (tm-luke) [inline Cloudmarque.Azure/Public/Deploy/IaaS/networking/vnetPeerings/New-CmAzIaasVnetPeerings.ps1:46]: Create vnet peerings

File: Cloudmarque.Azure/Public/Deploy/IaaS/networking/vnetPeerings/New-CmAzIaasVnetPeerings.ps1 [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+            $location = (Get-AzResourceGroup -Name $vnetPeerings[0].source.resourceGroup).location
+            $deploymentName = Get-CmAzResourceName -Resource "Deployment" -Architecture "IaaS" -Region $location -Name "New-CmAzIaasVnetPeerings"
+
+            New-AzDeployment `
+                -Name $deploymentName `
+                -TemplateFile $PSScriptRoot\New-CmAzIaasVnetPeerings.json `
+                -Location $location `
+                -VnetPeerings $vnetPeerings
+
+			Write-Verbose "Finished."
```
Comment (tm-luke) [inline Cloudmarque.Azure/Public/Deploy/IaaS/networking/vnetPeerings/New-CmAzIaasVnetPeerings.ps1:77]: Finished!

File: Cloudmarque.Azure/Resources/Project/Integration/vnetPeerings.yml [hunk 6]
```diff
@@ -0,0 +1,28 @@
+component: vnetPeerings
+
+peerings:
+  - source:
```
Comment (tm-luke) [inline Cloudmarque.Azure/Resources/Project/Integration/vnetPeerings.yml:4]: Remove indent for all arrays

File: Cloudmarque.Azure/Resources/Schema/New-CmAzIaasVnetPeerings.Schema.json [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+			"type": ["string", "null"],
+			"title": "Component",
+			"description": "Value to determine what cmdlet should be dynamically loaded for these settings."
+		},
+		"peerings": {
+			"$id": "#/peerings",
+			"type": "array",
+			"title": "Peerings",
+			"description": "Contains list of details required to deploy vnet peerings.",
+			"items": {
```
Comment (tm-luke) [inline Cloudmarque.Azure/Resources/Schema/New-CmAzIaasVnetPeerings.Schema.json:22]: Add minItems: 1 to item section

File: Cloudmarque.Azure/Resources/Schema/New-CmAzIaasVnetPeerings.Schema.json [hunk 8]
```diff
@@ -0,0 +1,130 @@
...
+                                    }
+								},
+								"additionalProperties": false
+							},
+							"targets": {
+								"$id": "#/peerings/items/allOf/targets",
+								"type": ["array"],
+								"title": "targets",
+								"description": "Contains list of target vnet details required to deploy vnet peerings with source vnet.",
+								"items": {
```
Comment (tm-luke) [inline Cloudmarque.Azure/Resources/Schema/New-CmAzIaasVnetPeerings.Schema.json:63]: Add minItems:1 to items section

File: Cloudmarque.Azure/Resources/Schema/New-CmAzIaasVnetPeerings.Schema.json [hunk 9]
```diff
@@ -0,0 +1,130 @@
...
+									"allOf": [
+										{
+											"$id": "#/peerings/items/allOf/targets/items/allOf",
+											"type": "object",
+											"required": [
+												"resourceGroup",
+                                                "vnetName"
+											],
+											"properties": {
+												"resourceGroup": {
```
Comment (tm-luke) [inline Cloudmarque.Azure/Resources/Schema/New-CmAzIaasVnetPeerings.Schema.json:74]: Isnt resourceGroup optional?

**Review by tm-luke — CHANGES_REQUESTED** [review]

File: Cloudmarque.Azure/Public/Common/Get-CmAzResourceName.ps1 (hunk 1 above)
Comment (deepend-dev) [reply Cloudmarque.Azure/Public/Common/Get-CmAzResourceName.ps1:94]: yes as it search in alias as well if the name doesnt match. that is how alias should be working earlier but that code was missing

File: Cloudmarque.Azure/Resources/Schema/New-CmAzIaasVnetPeerings.Schema.json (hunk 9 above)
Comment (deepend-dev) [reply Cloudmarque.Azure/Resources/Schema/New-CmAzIaasVnetPeerings.Schema.json:74]: nope

File: Cloudmarque.Azure/Public/Common/Get-CmAzResourceName.ps1 (hunk 1 above)
Comment (tm-luke) [reply Cloudmarque.Azure/Public/Common/Get-CmAzResourceName.ps1:94]: It should probably be $regions.regions | Where-Object { $_.name -Eq $Region -or $.alias -Eq $Region }

Comment (deepend-dev) [reply Cloudmarque.Azure/Public/Common/Get-CmAzResourceName.ps1:94]: done

File: Cloudmarque.Azure/Public/Deploy/IaaS/networking/vnetPeerings/New-CmAzIaasVnetPeerings.json (hunk 2 above)
Comment (deepend-dev) [reply Cloudmarque.Azure/Public/Deploy/IaaS/networking/vnetPeerings/New-CmAzIaasVnetPeerings.json]: done

File: Cloudmarque.Azure/Public/Deploy/IaaS/networking/vnetPeerings/New-CmAzIaasVnetPeerings.ps1 (hunk 3 above)
Comment (deepend-dev) [reply Cloudmarque.Azure/Public/Deploy/IaaS/networking/vnetPeerings/New-CmAzIaasVnetPeerings.ps1]: done

File: Cloudmarque.Azure/Public/Deploy/IaaS/networking/vnetPeerings/New-CmAzIaasVnetPeerings.ps1 (hunk 4 above)
Comment (deepend-dev) [reply Cloudmarque.Azure/Public/Deploy/IaaS/networking/vnetPeerings/New-CmAzIaasVnetPeerings.ps1]: done

File: Cloudmarque.Azure/Public/Deploy/IaaS/networking/vnetPeerings/New-CmAzIaasVnetPeerings.ps1 (hunk 5 above)
Comment (deepend-dev) [reply Cloudmarque.Azure/Public/Deploy/IaaS/networking/vnetPeerings/New-CmAzIaasVnetPeerings.ps1]: done

File: Cloudmarque.Azure/Resources/Project/Integration/vnetPeerings.yml (hunk 6 above)
Comment (deepend-dev) [reply Cloudmarque.Azure/Resources/Project/Integration/vnetPeerings.yml]: done

File: Cloudmarque.Azure/Resources/Schema/New-CmAzIaasVnetPeerings.Schema.json (hunk 7 above)
Comment (deepend-dev) [reply Cloudmarque.Azure/Resources/Schema/New-CmAzIaasVnetPeerings.Schema.json:22]: done

File: Cloudmarque.Azure/Resources/Schema/New-CmAzIaasVnetPeerings.Schema.json (hunk 8 above)
Comment (deepend-dev) [reply Cloudmarque.Azure/Resources/Schema/New-CmAzIaasVnetPeerings.Schema.json:64]: done

File: Cloudmarque.Azure/Resources/Schema/New-CmAzIaasVnetPeerings.Schema.json (hunk 9 above)
Comment (deepend-dev) [reply Cloudmarque.Azure/Resources/Schema/New-CmAzIaasVnetPeerings.Schema.json:75]: you are right. updated

**Review by tm-luke — APPROVED** [review]

**Pull request merged**